sign/script/common.js

```diff
@@ -0,0 +1,123 @@
+const formElement = document.querySelector('#form-list');
+
+const emailInput = document.querySelector('.email-form__input');
+const passwordInput = document.querySelector('.password-form__input');
+const passwordRepeatInput = document.querySelector('.password-repeat-form__input');
+
+const emailErrorText = document.querySelector('#email-error');
+const passwordErrorText = document.querySelector('#password-error');
+const passwordRepeatErrorText = document.querySelector('#password-repeat-error');
+
+const visibilityIconList = document.querySelectorAll('.switch-icon');
+
+const emailRegex =
+  /^(([^<>()\[\].,;:\s@"]+(\.[^<>()\[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i;
+const passwordRegex = /^(?=.*[a-zA-Z])(?=.*[0-9]).{8,25}$/;
+
+let isValidEmail = false;
+let isValidPassword = false;
+let isValidPasswordRepeat = false;
+
+// 회원가입 이메일 유효성 검사
+function validateSignUpEmail() {
+  if (!emailInput.value) {
+    // input 공백 시 실행
+    emailInput.classList.add('error-border');
+    emailErrorText.textContent = '이메일를 입력해주세요.';
+    isValidEmail = false;
+  } else if (emailRegex.test(emailInput.value) === false) {
+    // 이메일 유형이 아닐 시 실행
+    emailInput.classList.add('error-border');
+    emailErrorText.textContent = '올바른 이메일 주소가 아닙니다.';
+    isValidEmail = false;
+  } else if (emailInput.value === 'test@codeit.com') {
+    // test@codeit.com 입력 시 실행
+    emailInput.classList.add('error-border');
+    emailErrorText.textContent = '이미 사용 중인 이메일입니다.';
+    isValidEmail = false;
+  } else {
+    // 유효성 검사 통과
+    emailErrorText.textContent = '';
+    emailInput.classList.remove('error-border');
+    isValidEmail = true;
+  }
+}
```

Comment:
else if와 else 문이 계속해서 이어지네요. 

가독성을 더 좋게 하려면 Early return 패턴을 사용해서 리팩터링해보시는 것을 추천드립니다.

Early return 패턴은 코드의 가독성과 유지보수성을 향상시킬 수 있는 패턴 중 하나입니다. 코드에서 조건이 충족되지 않으면 함수의 실행을 즉시 종료하고, 조건이 충족될 때만 나머지 로직을 실행하는 방식입니다. 이 패턴을 사용하면 중첩된 if-else 구문을 피하고 코드의 흐름을 단순화할 수 있습니다.

```suggestion
function validateSignUpEmail() {
  const emailValue = emailInput.value;

  if (!emailValue) {
    // input 공백 시 실행
    emailInput.classList.add('error-border');
    emailErrorText.textContent = '이메일을 입력해주세요.';
    isValidEmail = false;
    return;
  }

  if (emailRegex.test(emailValue) === false) {
    // 이메일 유형이 아닐 시 실행
    emailInput.classList.add('error-border');
    emailErrorText.textContent = '올바른 이메일 주소가 아닙니다.';
    isValidEmail = false;
    return;
  }

  if (emailValue === 'test@codeit.com') {
    // test@codeit.com 입력 시 실행
    emailInput.classList.add('error-border');
    emailErrorText.textContent = '이미 사용 중인 이메일입니다.';
    isValidEmail = false;
    return;
  }

  // 유효성 검사 통과
  emailErrorText.textContent = '';
  emailInput.classList.remove('error-border');
  isValidEmail = true;
}
```

Reply:
위와 동일하게 수정 완료했습니다!! [https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/99/commits/961a6bcd9a904afae85cb71f4c214f4eec1d239b](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/99/commits/961a6bcd9a904afae85cb71f4c214f4eec1d239b), [https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/99/commits/879c4b7184429115da6a4939a3eaae656b828c1e](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/99/commits/879c4b7184429115da6a4939a3eaae656b828c1e)